pkg/destroy/aws/aws.go

```diff
@@ -289,12 +294,14 @@ func (o *ClusterUninstaller) findResourcesToDelete(
 		}
 	}
 
-	// Find IAM roles
-	iamRoleResources, err := findIAMRoles(ctx, iamRoleSearch, deleted, o.Logger)
-	if err != nil {
-		errs = append(errs, err)
+	// Find IAM roles if this is not an STS cluster
+	if o.CredentialsMode != string(types.ManualCredentialsMode) {
```

Comment:
`CredentialsMode` can be set to `Manual` without implying STS use: https://docs.openshift.com/container-platform/4.11/installing/installing_aws/manually-creating-iam.html#manually-creating-iam-aws. Do we also want to skip deleting IAM roles in that case?